In [4]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import glob

# === 1. Cargar features ===

#### 📂 Archivos que tienes

Cada archivo representa tráfico de red de un día/lapso específico con ataques o tráfico normal:

* Monday-WorkingHours.pcap_ISCX.csv → Solo tráfico normal (baseline).
* Tuesday-WorkingHours.pcap_ISCX.csv → Incluye tráfico normal y algunos ataques.
* Wednesday-workingHours.pcap_ISCX.csv → Normal + ataques (brute force, botnet, etc.).
* Thursday-WorkingHours-Morning-WebAttacks.pcap_ISCX.csv → Ataques de tipo web (XSS, SQL Injection, Brute Force).
* Thursday-WorkingHours-Afternoon-Infilteration.pcap_ISCX.csv → Ataques de infiltración.
* Friday-WorkingHours-Morning.pcap_ISCX.csv → Normal + algunos ataques.
* Friday-WorkingHours-Afternoon-PortScan.pcap_ISCX.csv → Ataques de PortScan.
* Friday-WorkingHours-Afternoon-DDos.pcap_ISCX.csv → Ataques de DDoS.

In [5]:
root_peth = '/home/javi/.cache/kagglehub/datasets/chethuhn/network-intrusion-dataset/versions/1/'

files = glob.glob(root_peth + "*.csv")

# Cargar y concatenar
df_list = [pd.read_csv(f) for f in files]
df = pd.concat(df_list, ignore_index=True)

(2830743, 79)


#### 📑 Descripción de columnas del dataset CIC-IDS-2017
| **Columna**                                                     | **Descripción**                                                                         |
| --------------------------------------------------------------- | --------------------------------------------------------------------------------------- |
| **Destination Port**                                            | Puerto de destino del flujo (ej. 80 para HTTP, 443 para HTTPS).                         |
| **Flow Duration**                                               | Duración total del flujo en microsegundos.                                              |
| **Total Fwd Packets**                                           | Número total de paquetes enviados desde origen → destino (forward).                     |
| **Total Backward Packets**                                      | Número total de paquetes enviados desde destino → origen (backward).                    |
| **Total Length of Fwd Packets**                                 | Suma de longitudes (bytes) de todos los paquetes forward.                               |
| **Total Length of Bwd Packets**                                 | Suma de longitudes (bytes) de todos los paquetes backward.                              |
| **Fwd Packet Length (Max/Min/Mean/Std)**                        | Estadísticas de tamaño de los paquetes forward.                                         |
| **Bwd Packet Length (Max/Min/Mean/Std)**                        | Estadísticas de tamaño de los paquetes backward.                                        |
| **Flow Bytes/s**                                                | Tasa de transmisión en bytes por segundo.                                               |
| **Flow Packets/s**                                              | Tasa de transmisión en paquetes por segundo.                                            |
| **Flow IAT (Mean/Std/Max/Min)**                                 | Estadísticas de tiempo entre llegada de paquetes en todo el flujo (Inter Arrival Time). |
| **Fwd IAT (Total/Mean/Std/Max/Min)**                            | Estadísticas de tiempos entre paquetes forward.                                         |
| **Bwd IAT (Total/Mean/Std/Max/Min)**                            | Estadísticas de tiempos entre paquetes backward.                                        |
| **Fwd PSH Flags / Bwd PSH Flags**                               | Número de flags TCP PSH (push) forward / backward.                                      |
| **Fwd URG Flags / Bwd URG Flags**                               | Número de flags TCP URG (urgent) forward / backward.                                    |
| **Fwd Header Length / Bwd Header Length**                       | Tamaño total de cabeceras en forward / backward.                                        |
| **Fwd Packets/s / Bwd Packets/s**                               | Tasa de paquetes forward / backward por segundo.                                        |
| **Min / Max / Mean / Std / Variance Packet Length**             | Estadísticas globales de tamaños de paquetes del flujo.                                 |
| **FIN / SYN / RST / PSH / ACK / URG / CWE / ECE Flag Count**    | Contadores de flags TCP observados en el flujo.                                         |
| **Down/Up Ratio**                                               | Relación entre el número de paquetes downstream y upstream.                             |
| **Average Packet Size**                                         | Tamaño promedio de los paquetes del flujo.                                              |
| **Avg Fwd Segment Size / Avg Bwd Segment Size**                 | Tamaño promedio de segmentos en forward / backward.                                     |
| **Fwd Header Length.1**                                         | Duplicado de `Fwd Header Length` (en algunos CSV).                                      |
| **Fwd Avg Bytes/Bulk, Fwd Avg Packets/Bulk, Fwd Avg Bulk Rate** | Métricas de ráfagas (bulk) de datos forward.                                            |
| **Bwd Avg Bytes/Bulk, Bwd Avg Packets/Bulk, Bwd Avg Bulk Rate** | Métricas de ráfagas (bulk) de datos backward.                                           |
| **Subflow Fwd Packets / Bytes**                                 | Paquetes y bytes en subflujos forward.                                                  |
| **Subflow Bwd Packets / Bytes**                                 | Paquetes y bytes en subflujos backward.                                                 |
| **Init_Win_bytes_forward / backward**                           | Tamaño de ventana inicial TCP anunciada en forward / backward.                          |
| **act_data_pkt_fwd**                                            | Número de paquetes con datos enviados en forward.                                       |
| **min_seg_size_forward**                                        | Tamaño mínimo de segmento en forward.                                                   |
| **Active (Mean/Std/Max/Min)**                                   | Estadísticas de intervalos en que el flujo estuvo activo (transmitiendo).               |
| **Idle (Mean/Std/Max/Min)**                                     | Estadísticas de intervalos en que el flujo estuvo inactivo (sin transmisión).           |
| **Label**                                                       | Clase del tráfico: `BENIGN` o tipo de ataque (`DDoS`, `PortScan`, `Web Attack`, etc.).  |


In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2830743 entries, 0 to 2830742
Data columns (total 79 columns):
 #   Column                        Dtype  
---  ------                        -----  
 0    Destination Port             int64  
 1    Flow Duration                int64  
 2    Total Fwd Packets            int64  
 3    Total Backward Packets       int64  
 4   Total Length of Fwd Packets   int64  
 5    Total Length of Bwd Packets  int64  
 6    Fwd Packet Length Max        int64  
 7    Fwd Packet Length Min        int64  
 8    Fwd Packet Length Mean       float64
 9    Fwd Packet Length Std        float64
 10  Bwd Packet Length Max         int64  
 11   Bwd Packet Length Min        int64  
 12   Bwd Packet Length Mean       float64
 13   Bwd Packet Length Std        float64
 14  Flow Bytes/s                  float64
 15   Flow Packets/s               float64
 16   Flow IAT Mean                float64
 17   Flow IAT Std                 float64
 18   Flow IAT Max         

In [12]:
df.columns.tolist()

[' Destination Port',
 ' Flow Duration',
 ' Total Fwd Packets',
 ' Total Backward Packets',
 'Total Length of Fwd Packets',
 ' Total Length of Bwd Packets',
 ' Fwd Packet Length Max',
 ' Fwd Packet Length Min',
 ' Fwd Packet Length Mean',
 ' Fwd Packet Length Std',
 'Bwd Packet Length Max',
 ' Bwd Packet Length Min',
 ' Bwd Packet Length Mean',
 ' Bwd Packet Length Std',
 'Flow Bytes/s',
 ' Flow Packets/s',
 ' Flow IAT Mean',
 ' Flow IAT Std',
 ' Flow IAT Max',
 ' Flow IAT Min',
 'Fwd IAT Total',
 ' Fwd IAT Mean',
 ' Fwd IAT Std',
 ' Fwd IAT Max',
 ' Fwd IAT Min',
 'Bwd IAT Total',
 ' Bwd IAT Mean',
 ' Bwd IAT Std',
 ' Bwd IAT Max',
 ' Bwd IAT Min',
 'Fwd PSH Flags',
 ' Bwd PSH Flags',
 ' Fwd URG Flags',
 ' Bwd URG Flags',
 ' Fwd Header Length',
 ' Bwd Header Length',
 'Fwd Packets/s',
 ' Bwd Packets/s',
 ' Min Packet Length',
 ' Max Packet Length',
 ' Packet Length Mean',
 ' Packet Length Std',
 ' Packet Length Variance',
 'FIN Flag Count',
 ' SYN Flag Count',
 ' RST Flag Count',
 ' P

In [14]:
df[' Label'].value_counts()

 Label
BENIGN                        2273097
DoS Hulk                       231073
PortScan                       158930
DDoS                           128027
DoS GoldenEye                   10293
FTP-Patator                      7938
SSH-Patator                      5897
DoS slowloris                    5796
DoS Slowhttptest                 5499
Bot                              1966
Web Attack � Brute Force         1507
Web Attack � XSS                  652
Infiltration                       36
Web Attack � Sql Injection         21
Heartbleed                         11
Name: count, dtype: int64

📑 Clases del dataset CIC-IDS-2017
| **Clase**                      | **Cantidad** | **Descripción**                                                                                                                          |
| ------------------------------ | ------------ | ---------------------------------------------------------------------------------------------------------------------------------------- |
| **BENIGN**                     | 2,273,097    | Tráfico normal sin ataques (baseline de comparación).                                                                                    |
| **DoS Hulk**                   | 231,073      | Ataque **Denial of Service (DoS)** que envía un gran volumen de tráfico HTTP para saturar el servidor.                                   |
| **PortScan**                   | 158,930      | Escaneo masivo de puertos para descubrir servicios abiertos en un host.                                                                  |
| **DDoS**                       | 128,027      | **Distributed DoS**, múltiples hosts atacan a un objetivo simultáneamente para sobrecargarlo.                                            |
| **DoS GoldenEye**              | 10,293       | Ataque DoS contra servidores web mediante múltiples solicitudes HTTP incompletas.                                                        |
| **FTP-Patator**                | 7,938        | Ataque de fuerza bruta contra servicio **FTP** (intentos repetidos de login con diferentes credenciales).                                |
| **SSH-Patator**                | 5,897        | Ataque de fuerza bruta contra servicio **SSH**.                                                                                          |
| **DoS slowloris**              | 5,796        | Ataque DoS que mantiene múltiples conexiones HTTP abiertas enviando cabeceras parciales para agotar recursos del servidor.               |
| **DoS Slowhttptest**           | 5,499        | Ataque DoS similar a slowloris, que envía peticiones HTTP lentas para consumir conexiones del servidor.                                  |
| **Bot**                        | 1,966        | Tráfico generado por dispositivos comprometidos que forman parte de una **botnet**.                                                      |
| **Web Attack – Brute Force**   | 1,507        | Ataque web que intenta adivinar contraseñas mediante intentos repetidos de login.                                                        |
| **Web Attack – XSS**           | 652          | **Cross-Site Scripting (XSS)**: inyección de scripts maliciosos en páginas web.                                                          |
| **Infiltration**               | 36           | Acceso no autorizado a una red interna desde un host comprometido.                                                                       |
| **Web Attack – SQL Injection** | 21           | Inyección de código SQL malicioso en formularios o peticiones web para manipular bases de datos.                                         |
| **Heartbleed**                 | 11           | Explotación de la vulnerabilidad **Heartbleed (CVE-2014-0160)** en OpenSSL para filtrar información sensible de la memoria del servidor. |
